WooCommerce/Classes/Analytics/WooAnalytics.swift

```diff
@@ -61,7 +61,11 @@ public extension WooAnalytics {
             return
         }
 
-        analyticsProvider.refreshUserData()
+        // Skips refreshing user data when user is authenticated without WPCom
+        // since they are still identified with anonymous ID.
+        if ServiceLocator.stores.isAuthenticatedWithoutWPCom == false {
```

Comment:
Nit: Consider injecting `stores` to unit test this case.

Reply:
I tried adding `stores` in the intializer of `WooAnalytics` but it causes a recursion.